integration-tests/test/fee-payment.spec.ts

```diff
@@ -36,17 +41,30 @@ describe('Fee Payment Integration Tests', async () => {
   it('Paying a nonzero but acceptable gasPrice fee', async () => {
     const amount = utils.parseEther('0.5')
     const balanceBefore = await env.l2Wallet.getBalance()
+    const feeVaultBalanceBefore = await env.l2Wallet.provider.getBalance(
+      predeploys.OVM_SequencerFeeVault
+    )
     expect(balanceBefore.gt(amount))
 
     const tx = await env.ovmEth.transfer(other, amount)
     const receipt = await tx.wait()
     expect(receipt.status).to.eq(1)
 
     const balanceAfter = await env.l2Wallet.getBalance()
+    const feeVaultBalanceAfter = await env.l2Wallet.provider.getBalance(
+      predeploys.OVM_SequencerFeeVault
+    )
+    const expectedFeePaid = tx.gasPrice.mul(tx.gasLimit)
+
     // The fee paid MUST be the receipt.gasUsed, and not the tx.gasLimit
     // https://github.com/ethereum-optimism/optimism/blob/0de7a2f9c96a7c4860658822231b2d6da0fefb1d/packages/contracts/contracts/optimistic-ethereum/OVM/accounts/OVM_ECDSAContractAccount.sol#L103
-    expect(balanceBefore.sub(balanceAfter)).to.be.deep.eq(
-      tx.gasPrice.mul(tx.gasLimit).add(amount)
+    expect(balanceBefore.sub(balanceAfter)).to.deep.equal(
+      expectedFeePaid.add(amount)
+    )
+
+    // Make sure the fee was transferred to the vault.
+    expect(feeVaultBalanceAfter.sub(feeVaultBalanceBefore)).to.deep.equal(
+      expectedFeePaid
```

Comment:
It would be a good idea to add integration test coverage on the withdrawal of fees to L1 as well. Would this be super tricky? I'm trying to think whether this would be much harder than our normal withdrawal tests but it seems like it would be doable.

Reply:
Nah should be doable.